scripts/forge-script.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+set -eo pipefail
+
+source "${BASH_SOURCE%/*}/_common.sh"
+
+deploy() {
+  normalize-env-vars
+
+  local PASSWORD="$(extract-password)"
+  local PASSWORD_OPT=()
+  if [ -n "$PASSWORD" ]; then
+    PASSWORD_OPT=(--password "$PASSWORD")
+  fi
+
+  check-required-etherscan-api-key
+
+  local RESPONSE=
+  # Log the command being issued, making sure not to expose the password
+  log "forge script --broadcast --slow --verify --retries 10 --json --sender $ETH_FROM --rpc-url $ETH_RPC_URL --gas-price $ETH_GAS --gas-limit $FOUNDRY_GAS_LIMIT --keystores="$FOUNDRY_ETH_KEYSTORE_FILE" $(sed 's/ .*$/ [REDACTED]/' <<<"${PASSWORD_OPT[@]}")" $(printf ' %q' "$@")
+  # Currently `forge script` sends the logs to stdout instead of stderr.
+  # This makes it hard to compose its output with other commands, so here we are:
+  # 1. Duplicating stdout to stderr through `tee`
+  # 2. Extracting only the address of the deployed contract to stdout
+  RESPONSE=$(forge script --broadcast --slow --verify --retries 10 --json --sender $ETH_FROM --rpc-url $ETH_RPC_URL --gas-price $ETH_GAS --gas-limit $FOUNDRY_GAS_LIMIT --keystores="$FOUNDRY_ETH_KEYSTORE_FILE" "${PASSWORD_OPT[@]}" "$@" | tee >(cat 1>&2))
+  jq -R 'fromjson? | .logs | .[]' <<<"$RESPONSE" | xargs -I@ cast --to-ascii @ | jq -R 'fromjson?' | jq -s 'map( {(.[0]): .[1]} ) | add'
```

Comment:
I'd move this `jq` line to the `src/RWA007-A/scripts/deploy.sh` deploy file, otherwise we'd be implicitly expecting that any scripts inside this repo to log deployed contracts in this format.

However, we might need scripts to do different things, such as amending permissions or collecting data.

I would also do the same with the `--broadcast --slow --verify --retries 10` flags: pass them in the caller, not hard-code them here.

Reply:
done